Added a service to xor `Bytes`

ppepos opened the pull request:
Needed a new constructor for `Bytes` to facilitate things.

privat commented on lib/core/bytes.nit:
```diff
@@ -173,6 +173,17 @@ class Bytes
 	end
 
+	# Init a `Bytes` from a string
+	init from(s: Text) do
```
Why not using `Text::to_bytes`?

privat commented on lib/crypto.nit:
```diff
@@ -183,6 +183,36 @@ redef class Text
+	# Returns `self` xored with `key`
+	#
+	# The shortest of the two is cycled through until the longest has been
+	# completely xored.
```
I'm not sure if using the shortest is a POLA approach.
I assumed that xorcrypting was symmetric. ie `message.xor(key).xor(key) == message`. But if key.length > message.length this become false.

privat replied:
also rename the method `xorcipher` or else people will mistake it with the mathematical bitwise xor operation (ie without cycling).

ppepos commented:
* Removed useless constructor.
* Added https://github.com/nitlang/nit/pull/2061/commits/2580cc81efbb38638f26b0d6c16630e7a981e1f2
* Changed algorithm to not pick longest as `message` and shortest as `key` but to always consider `self`as `message`.

privat commented:
+1

jcbrinfo commented:
:+1: +1
